File: Testing/Code for Testing/Early Basic Syntax Testing/ffi/test_ffi.js
const sqrt = Math.sqrt;
function puts(s) { console.log(s); return 0; }


function main() {
    let val = 144.0;
    let result = sqrt(val);
    console.log(result);
    puts("Hello from FFI!");
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();

File: Testing/Code for Testing/Early Basic Syntax Testing/ffi/test_ffi.py
from dataclasses import dataclass, field
from math import sqrt
def puts(s): print(s); return 0

def _ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class _ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)


def main():
    val = 144.0
    result = sqrt(val)
    print(result)
    puts('Hello from FFI!')
    return 0


if __name__ == "__main__":
    main()
